judge/views/problem.py

```diff
@@ -619,25 +619,32 @@ def form_valid(self, form):
                                    _('You have exceeded the submission limit for this problem.'))
 
         with transaction.atomic():
-            self.new_submission = form.save()
+            self.new_submission = form.save(commit=False)
 
             contest_problem = self.contest_problem
             if contest_problem is not None:
-                # See ContestSubmission post_save hook in judge/signals.py
-                self.new_submission.contest_object_id = contest_problem.contest_id
+                # Use the contest object from current_contest.contest because we already use it
+                # in profile.update_contest().
+                # See ContestSubmission post_save hook in judge/signals.py.
+                self.new_submission.contest_object = self.request.profile.current_contest.contest
+                if self.request.profile.current_contest.live:
+                    self.new_submission.locked_after = self.new_submission.contest_object.locked_after
+                self.new_submission.save()
                 ContestSubmission(
                     submission=self.new_submission,
-                    problem=contest_problem,
+                    problem=self.contest_problem,
```

Comment:
Is there any point to adding `self.` here?

Reply:
Oops, was testing something and forgot to revert this :facepalm:. Fixed.